src/main/java/bio/terra/service/profile/ProfileApiController.java

```diff
@@ -173,4 +180,12 @@ private void verifyProfileAuthorization(
     // Verify profile permissions
     iamService.verifyAuthorization(userReq, resourceType, resourceId, action);
   }
+
+  private void verifyAdminAuthorization(AuthenticatedUserRequest userReq) {
+    iamService.verifyAuthorization(
+        userReq,
+        IamResourceType.DATAREPO,
+        applicationConfiguration.getResourceId(),
+        IamAction.CONFIGURE);
```

Comment:
I think we would then need to wire in ApplicationConfiguration to IamService to get the resource Id. Given that we only do this admin check in a few places, would it be worth pull through applicationConfiguration everywhere that we use call iamservice? But, maybe I'm misunderstanding what would be required!

Reply:
Yes, moving it into IamService would require wiring ApplicationConfiguration into it. It wouldn't change the way IamService APIs are called but it would add a new dependency so it might not be worth doing.

The thing I was thinking was that TDR uses a check on the `CONFIGURE` action as a proxy for an admin user check, and that's the thing I wanted to clean up. But now that I look at uses of `ApplicationConfiguration.getResourceId()` I see that there's a couple different actions that are checked. Maybe this is a ticket for a future cleanup; it seems like if we're using `CONFIGURE` as a proxy for `is admin user`, it should really be `ADMIN` or something else. Either that, or we should have different actions for different admin type operations, such as `SYNC_DUOS_USERS`.

Reply:
Checking the "configure" action as a proxy for the admin check also provides a confusing error message for the user. (Rather than saying you need to be an admin to complete the action, it just says that your user doesn't have the 'configure' action). 
Ticket entered here: https://broadworkbench.atlassian.net/browse/DR-3248